pkg/componentreadiness/tracker/tracker.go

```diff
@@ -0,0 +1,280 @@
+package tracker
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/google/uuid"
+	"github.com/openshift/sippy/pkg/apis/api"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+)
+
+const (
+	testRegressionsDataSet = "ci_analysis_us"
+	testRegressionsTable   = "test_regressions"
+)
+
+type Storage interface {
+	ListCurrentRegressions(release string) ([]api.TestRegression, error)
+	OpenRegression(release string, newRegressedTest api.ComponentReportTestSummary) (*api.TestRegression, error)
+	ReOpenRegression(regressionID string) error
+	CloseRegression(regressionID string, closedAt time.Time) error
+}
+
+// BigQueryStorage is the primary implementation for real world usage, tracking when we detect test
+// regressions opening/closing in the Bigquery test_regressions table.
+type BigQueryStorage struct {
+	client *bigquery.Client
+}
+
+func NewBigQueryStorage(client *bigquery.Client) Storage {
+	return &BigQueryStorage{client: client}
+}
```

Comment:
Naming things is hard. Big Query Storage is the name of a bigquery API, I think we should pick a different name to avoid overloading. This is in a directory called tracker.go, I'd expect see things related to the word "Tracking" in here.

Maybe RegressionTracker, and the impl BigQueryRegressionTracker?

Reply:
Good point I'll adjust to something like that.

Reply:
I wanted to keep an underlying interface for abstracting away bigquery (for testing purposes... someday), so that RegressionTracker was already claimed, but I moved this to RegressionStore + BigQueryRegressionStore. Hopefully that helps.